pytorch_lightning/callbacks/progress/base.py

```diff
@@ -48,13 +48,28 @@ def on_train_batch_end(self, trainer, pl_module, outputs, batch_idx):
 
     def __init__(self) -> None:
         self._trainer: Optional["pl.Trainer"] = None
+        self._current_eval_dataloader_idx: Optional[int] = None
```

Comment:
well, for each dataloader, even the first one, it needs to be assigned inside batch_start hook, now if I make it as 0, then I would have to keep an additional initialization inside `on_validation_epoch_start` hook, and we will have the bar init at 2 different places. If I keep it None then `None!=0` condition will satisfy and it will create the bar accordingly.

Also if FT is initialized with dataloader_idx=1, then default 0 won't work.

Reply:
If you initialize to 0, what's the problem with the following in a batch start hook?

```python
    if self.bar is None or self.current_dl_idx != dl_idx:
        self.bar = Bar()
```

Reply:
won't work if someone calls `trainer.fit` and then `trainer.validate`.

else we will have to go with 
```py
if self.bar is None or self.bar.is_closed_or_something or self.current_dl_idx != dl_idx
```
but doing this adds more responsibility on a single hook.
my idea here was to initialize the bar in epoch_start hooks and initialize the counters in batch_start hooks.